theano/gpuarray/dnn_redux.c

```diff
@@ -61,11 +61,6 @@ int APPLY_SPECIFIC(dnn_redux)(PyGpuArrayObject *input,
   static float fbeta = 0.0f;
   static double dbeta = 0.0;  
 
-  if (!GpuArray_IS_C_CONTIGUOUS(&input->ga)) {
```

Comment:
This would need tests, as it is not part of the cudnn doc. So even if it work in one cases, it don't mean it will work with all cases. Also futur version can change this.

Until we get an official confirmation that this is safe, we should add back this restriction and add a gpu_contiguous in the opt that move the code to the GPU.